packages/assets-controllers/src/TokensController.ts

```diff
@@ -697,53 +716,66 @@ export class TokensController extends BaseController<
    * @param params.newTokens - The new tokens to set for the current network and selected account.
    * @param params.newIgnoredTokens - The new ignored tokens to set for the current network and selected account.
    * @param params.newDetectedTokens - The new detected tokens to set for the current network and selected account.
+   * @param params.detectionAddress - The address on which the detected tokens to add were detected.
+   * @param params.detectionChainId - The chainId on which the detected tokens to add were detected.
    * @returns The updated `allTokens` and `allIgnoredTokens` state.
    */
   _getNewAllTokensState(params: {
     newTokens?: Token[];
     newIgnoredTokens?: string[];
     newDetectedTokens?: Token[];
+    detectionAddress?: string;
+    detectionChainId?: string;
   }) {
-    const { newTokens, newIgnoredTokens, newDetectedTokens } = params;
+    const {
+      newTokens,
+      newIgnoredTokens,
+      newDetectedTokens,
+      detectionAddress,
+      detectionChainId,
+    } = params;
     const { allTokens, allIgnoredTokens, allDetectedTokens } = this.state;
     const { chainId, selectedAddress } = this.config;
 
+    const userAddressToAddTokens = detectionAddress ?? selectedAddress;
+    const chainIdToAddTokens = detectionChainId ?? chainId;
+
     let newAllTokens = allTokens;
-    if (newTokens) {
-      const networkTokens = allTokens[chainId];
+    if (newTokens?.length) {
```

Comment:
added this because if, for example, `newTokens === []` we would still enter this if block and add an empty array to the chainId/selectedAddress combo in `newAllTokens`